### ISSUE TITLE
sdks/cpp needs unit tests

### ISSUE THREAD

**Issue by devjgm** [googleforgames/agones#905]
**Is your feature request related to a problem? Please describe.**

The C++ code in sdks/cpp/ doesn't have any unit tests, so there's no way to verify that an of it works. This makes it difficult for anyone to contributed a change/fix to the code base, because there's no way to prove that the code works (before or) after the change.

**Describe the solution you'd like**

All the C++ files in sdks/cpp should have a `foo_test.cc` that corresponds to `foo.h` and `foo.cc`. The unit tests should be implemented using Google Test (https://github.com/google/googletest)

**Describe alternatives you've considered**

None. There's no alternative to unit tests :-)

**Additional context**

There are other unit testing frameworks, but I believe Google Test is by far the most common for C++, and it's the one required for Google C++ code.

**Comment by markmandel** [comment]
Yes please! We definitely need someone with C++ experience to write this 👍

**Comment by markmandel** [comment]
We should probably do this at some point.

**Comment by markmandel** [comment]
Adding `awainting-maintainer` label, since we really should do this.

**Comment by zedGGs** [comment]
Heey @markmandel, would be good idea to add required tests, 
I can do something about that, if u think that's okey u can assign me

**Comment by markmandel** [comment]
@zedGGs please go for it! Consider yourself assigned. Let us know if you have any questions here about the framework in place for SDKs, or in #development on Slack!

**Comment by zedGGs** [comment]
@markmandel awesome stuff ! thanks, sure,
Im gonna add pr !
